pkg/services/sqlstore/user_test.go

```diff
@@ -63,6 +64,53 @@ func TestUserDataAccess(t *testing.T) {
 			})
 		})
 
+		Convey("Given an organization", func() {
+			setting.AutoAssignOrg = true
+
+			orgCmd := &models.CreateOrgCommand{Name: "Some Test Org"}
+			err := CreateOrg(orgCmd)
+			So(err, ShouldBeNil)
+
+			Convey("Creates user assigned to other organization", func() {
+				cmd := &models.CreateUserCommand{
+					Email: "usertest@test.com",
+					Name:  "user name",
+					Login: "user_test_login",
+					OrgId: orgCmd.Result.Id,
+				}
+
+				err := CreateUser(context.Background(), cmd)
+				So(err, ShouldBeNil)
+
+				Convey("Loading a user", func() {
+					query := models.GetUserByIdQuery{Id: cmd.Result.Id}
+					err := GetUserById(&query)
+					So(err, ShouldBeNil)
+
+					So(query.Result.Email, ShouldEqual, "usertest@test.com")
+					So(query.Result.Password, ShouldEqual, "")
+					So(query.Result.Rands, ShouldHaveLength, 10)
+					So(query.Result.Salt, ShouldHaveLength, 10)
+					So(query.Result.IsDisabled, ShouldBeFalse)
+					So(query.Result.OrgId, ShouldEqual, orgCmd.Result.Id)
+				})
+			})
+
+			Convey("Don't create user assigned to unknown organization", func() {
+				cmd := &models.CreateUserCommand{
+					Email: "usertest@test.com",
+					Name:  "user name",
+					Login: "user_test_login",
+					OrgId: 10000,
+				}
+
+				err := CreateUser(context.Background(), cmd)
+				So(err, ShouldNotBeNil)
+			})
+
+			setting.AutoAssignOrg = false
```

Comment:
@papagian agree, but but maybe it would be better to restore to the original value of `setting.AutoAssignOrg`, just in case it's `false`? Assumptions can be dangerous.

Reply:
good point, that would be even better.